src/librustc/hir/lowering.rs

```diff
@@ -2842,11 +2844,21 @@ impl<'a> LoweringContext<'a> {
                 self.lower_ty(t, ImplTraitContext::disallowed()),
                 self.lower_generics(generics, ImplTraitContext::disallowed()),
             ),
-            ItemKind::Existential(ref b, ref generics) => hir::ItemKind::Existential(hir::ExistTy {
-                generics: self.lower_generics(generics, ImplTraitContext::disallowed()),
-                bounds: self.lower_param_bounds(b, ImplTraitContext::disallowed()),
-                impl_trait_fn: None,
-            }),
+            ItemKind::Existential(ref b, ref generics) => {
+                let generics = self.lower_generics(generics, ImplTraitContext::disallowed());
+
+                let definitions = self.resolver.definitions();
+                let parent_node_id = self.current_hir_id_owner.get(self.current_hir_id_owner.len() - 1).map(|i|
+                    definitions.def_index_to_node_id(i.0));
+                let bounds = self.lower_param_bounds(b,
+                    ImplTraitContext::Existential(None, parent_node_id));
```

Comment:
Because associated type bindings can appear in bounds, but not generics. i.e. you can have `existential type Foo = Iterator<Item = impl Clone>;`, but `existential type Foo<Item = impl Clone> = ...;` just makes no sense.